src/runtime/action/index.ts

```diff
@@ -0,0 +1,8 @@
+export interface ActionReturn<Parameters = any> {
+	update?: (parameters: Parameters) => void;
+	destroy?: () => void;
+}
+
+export interface Action<Parameters = any> {
+	(node: HTMLElement, parameters: Parameters): void | ActionReturn<Parameters>;
```

Comment:
Do we want a generic for `node` type too, in case someone wants something more specific than `HTMLElement`.

Reply:
True, that would be good as a second generic parameter. I'm not sure if it should be first or second in the order it appears in the generics. First would be consistent with the order of the parameters, but it's arguably used less often.

Reply:
Having a generic would be nice for additional constraint and having it consistent with the order of parameters would also be nice and predictable as well, but it wouldn't be as a nice experience having to pass in the targeted node/element every time the parameters need to be defined.

With this consideration, I've placed it as the second generic argument so most users can ignore it since it's rarely used, as well as relaxing the constraint with type inference and extension, so the second generic can be more of a minimum requirement for the node instead of a fixed one.

Reply:
Putting it as the second argument makes sense to me 👍